sdk/nextjs/vercel.ts

```diff
@@ -0,0 +1,98 @@
+import { unstable_flag as flag } from '@vercel/flags/next'
+import { setupDevCycle } from './src/server/setupDevCycle'
+
+// mimic the Vercel JsonValue type since it is not exported
+type JsonValue =
+    | string
+    | number
+    | boolean
+    | { [key: string]: JsonValue | JsonValue[] }
+
+export const setupDevCycleVercelFlagHelper = (
+    context: Pick<
+        ReturnType<typeof setupDevCycle>,
+        'getConfig' | 'getAllFeatures'
+    >,
+) => {
+    return async <T extends JsonValue>(
+        key: string,
+        defaultValue: T,
+    ): Promise<T> => {
+        const { config } = await context.getConfig()
+        const variable = config.variables.find(
+            (variable) => variable.key === key,
+        )
+
+        const variableId = variable?._id
+        if (!variableId) {
+            console.error('[DevCycle] No Variable found for key:', key)
+            return defaultValue
+        }
+
+        const featureForVariable = config.features.find((feature) => {
+            return !!feature.variations[0]?.variables.find(
+                (variable) => variable._var === variableId,
+            )
+        })
+
+        if (!featureForVariable) {
+            console.error('[DevCycle] No Feature found for Variable', key)
+            return defaultValue
+        }
+
+        const getFlag = flag<string | null>({
+            key: featureForVariable.key,
+            defaultValue: null,
```

Comment:
why no default here?

Reply:
because this is serving the variation of the feature that should be used to obtain the flag value. The default for that should be "no variation" which is interpreted by the code further down as "use the variable's default"